src/lib/swapper/api.ts

```diff
@@ -274,3 +282,24 @@ export interface Swapper<T extends ChainId, MaybeUnknownNetworkFee extends boole
    */
   getTradeTxs(tradeResult: TradeResult): Promise<Result<TradeTxs, SwapErrorRight>>
 }
+
+export type Swapper3 = {
+  getUnsignedTx(
+    tradeQuote: TradeQuote & { receiveAddress: string; affiliateBps: string },
+    // Note, `wallet` isn't going to cut it once this actually lives as an API - added for compatibility with current chain-adapters implementation
+    // adapter.buildSendTransaction() should be able to sign a Tx, without a wallet
+    // We will want to decouple building a "send trasaction" (i.e a Tx to sign with a few additional fields), and the actual unsigned Tx building in chain-adapters
+    // Alternatively, since wallet is used for chain support detection and getAddress() calls, we could build a `buildUnsignedTx` variant
+    // with a slightly different payload
+    wallet: HDWallet,
+    // Similarly, this is for backwards compatibility. Do we want to only return the unsigned Tx, without the additional data here?
+    // This is very hdwallet/chain-adapters specific, and won't catter to the needs of consumers
+    chainId?: ChainId,
+    accountMetadata?: AccountMetadata,
```

Comment:
That's also most likely going to change, obviously not friendly to external consumers but fits our current implementation